### PR TITLE
[WIP] yt-4.0 added refined status and cell sizes to cyoctree 

### PR DESCRIPTION
**Pull request by AshKelly**
## PR Summary

Added the refined field, and the size of cells in various dimensions to the octree
e.g

```
octree = ...
```
in the case of 1 oct, refined into 8 leaves,
```
octree[('index', 'refined')]
```
returns
```
[True, False, False, False, False, False, False, False, False]
```

**this needs tests and docs**

## PR Checklist

- [ ] Code passes flake8 checker
- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/utilities/lib/cyoctree.pyx [hunk 1]
```diff
@@ -360,6 +360,66 @@ cdef class CyOctree:
...
+        cdef np.int8_t[:] refined = np.zeros(
+                (num_leaves*self._num_cells + self.nodes.size()),
+                dtype='int8'
+        )
+
+        z = 0
+        for i in range(self.nodes.size()):
+            if self.nodes[i].leaf == 0:
+                refined[z] = 1
+                z += 1
```
Comment (matthewturk) [inline yt/utilities/lib/cyoctree.pyx]: Only comment would be to remove the `else` and add a `continue` here so we drop the indentation.  But not a big deal.

**Comment by matthewturk** [comment]
@AshKelly This is great -- what else needs to happen for this to go in?

**Comment by AshKelly** [comment]
I need to add some tests and then I'll add the `volume` field which will fix #2279

**Comment by matthewturk** [comment]
@AshKelly Anything we can do to help get this over the finish line?

**Comment by dnarayanan** [comment]
@AshKelly i just checked and is it the case that #2279 still returns the "max recursion depth exceeded?"?

i'm on the `desika_octree` branch of your fork

**Pull request closed without merging**